### PR TITLE
New Plugin: Authorpages

### PR DESCRIPTION
**Pull request by dcrosta**
Requesting feedback on the approach. This basically works right now, but I'm not familiar enough with Pelican to know if I've implemented everything properly, or if I've missed something. All feedback appreciated.

### CODE REVIEW

File: authorpages/authorpages.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from pelican import signals
+
+
+def yield_files(root):
+    root = os.path.realpath(os.path.abspath(root))
+    for dirpath, dirnames, filenames in os.walk(root):
+        for dirname in list(dirnames):
+            try:
+                if dirname.startswith("."):
+                    dirnames.remove(dirname)
```
Comment (kylef) [inline authorpages/authorpages.py]: Not sure what the purpose of this is, it seems `dirnames` is never used?

Comment (dcrosta) [inline authorpages/authorpages.py]: If you mutate dirnames (at least by removing elements, possibly also by rearranging, etc) then os.walk respects those changes as it continues to iterate the file system. Here in just trying to skip dot-dirs (if any) for efficiency.

Comment (kylef) [inline authorpages/authorpages.py]: Interesting, I didn't know this. :+1: thanks.

Comment (kylef) [inline authorpages/authorpages.py]: Interesting, I didn't know this. Nice :+1:

**Comment by justinmayer** [comment]
Thanks for the contribution, Dan.

**Comment by pxquim** [comment]
Very interesting... can we have a similar solution for blog categories? ... and perhaps even blog tags as well?

**Comment by dcrosta** [comment]
@pxquim I don't see why not.

I kind of wish this were a Pelican core feature. Do the core devs have any interest in that? I can probably find some time to try to patch it in there (with support for categories, tags, and authors).

**Comment by avaris** [comment]
I have a plugin in development to handle a similar function for authors, categories and tags for a while now. Unfortunately, pesky real-life issues tend to get in the way. 

Putting it into the core is also a possibility, although I'm sort of against it because I'm in favor of a minimalistic core and a plugin is enough for this task.

**Comment by dcrosta** [comment]
@avaris I just put up #482 which extends the idea here to categories and tags -- you might want to take a look.

**Comment by avaris** [comment]
@drcosta I'm a bit late, it's merged but took a look anyway. 

It's similar to what I'm doing but there are some stuff you might want to watch. 

Authors, categories and tags are not actually singletons. Pelican creates a new object for these types for each article even if there was one with same `name`. `generator.categories` for example would contain only one of them and you're setting `page` for that one. This is mostly OK, but say one wanted to put that stuff in `article.html` template and tried to access `article.category.page`, it might result in an error.

This is not an issue because of the way you set up, but might want to reconsider. I set my paths relative to the `content` so I had to extend `ARTICLE_EXCLUDES` and `PAGE_EXCLUDES` with the paths in order to not create real content from them. Putting these files out of `content` doesn't feel right to me.

Again, this might not be an issue if it's your intention. You're using the `Page` object from core and it'll fire `content_object_init` signal which might trigger some other plugin that's expecting a regular content object. Just something to be careful.

**Comment by dcrosta** [comment]
@avaris thanks, I hadn't considered this:

> You're using the Page object from core and it'll fire content_object_init signal which might trigger some other plugin that's expecting a regular content object. Just something to be careful.

Any suggestions? I wanted to re-use as much code from core as I could, but I don't know core very well yet (I've only browsed the bits that seemed relevant, haven't read it all carefully).

**Pull request merged**